compiler-utils/src/testFixtures/java/com/sqareup/anvil/compiler/internal/testing/TestUtils.kt

```diff
@@ -0,0 +1,181 @@
+import com.squareup.anvil.compiler.AnvilComponentRegistrar
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.KotlinCompilation.Result
+import com.tschuchort.compiletesting.PluginOption
+import com.tschuchort.compiletesting.SourceFile
+import dagger.Component
+import dagger.Module
+import dagger.Subcomponent
+import dagger.internal.codegen.ComponentProcessor
+import org.jetbrains.kotlin.config.JvmTarget
+import java.io.File
+import java.io.OutputStream
+import java.util.Locale
+import kotlin.reflect.KClass
+
+/**
+ * Helpful for testing code generators in unit tests end to end.
+ */
+@ExperimentalAnvilApi
+public fun compileAnvil(
+  vararg sources: String,
+  enableDaggerAnnotationProcessor: Boolean = false,
+  generateDaggerFactories: Boolean = false,
+  generateDaggerFactoriesOnly: Boolean = false,
+  disableComponentMerging: Boolean = false,
+  allWarningsAsErrors: Boolean = true,
+  useIR: Boolean = true,
+  messageOutputStream: OutputStream = System.out,
+  block: Result.() -> Unit = { }
+): Result {
+  return KotlinCompilation()
+    .apply {
+      compilerPlugins = listOf(AnvilComponentRegistrar())
+      this.useIR = useIR
+      useOldBackend = !useIR
+      inheritClassPath = true
+      jvmTarget = JvmTarget.JVM_1_8.description
+      verbose = false
+      this.allWarningsAsErrors = allWarningsAsErrors
+      this.messageOutputStream = messageOutputStream
+
+      if (enableDaggerAnnotationProcessor) {
+        annotationProcessors = listOf(ComponentProcessor())
+      }
+
+      val anvilCommandLineProcessor = AnvilCommandLineProcessor()
+      commandLineProcessors = listOf(anvilCommandLineProcessor)
+
+      pluginOptions = listOf(
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "src-gen-dir",
+          optionValue = File(workingDir, "build/anvil").absolutePath
+        ),
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "generate-dagger-factories",
+          optionValue = generateDaggerFactories.toString()
+        ),
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "generate-dagger-factories-only",
+          optionValue = generateDaggerFactoriesOnly.toString()
+        ),
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "disable-component-merging",
+          optionValue = disableComponentMerging.toString()
+        )
+      )
+
+      this.sources = sources.map { content ->
+        val packageDir = content.lines()
+          .firstOrNull { it.trim().startsWith("package ") }
+          ?.substringAfter("package ")
+          ?.replace('.', '/')
+          ?.let { "$it/" }
+          ?: ""
+
+        val name = "${workingDir.absolutePath}/sources/src/main/java/$packageDir/Source.kt"
+        with(File(name).parentFile) {
+          check(exists() || mkdirs())
```

Comment:
Shorthand for make directory if not exists?

Reply:
Yes, pretty much :)